src/writers.rs

```diff
@@ -197,26 +198,24 @@ impl fmt::Display for EnvSection<'_> {
         write!(
             &mut separated.ready(),
             "{}",
-            SpanTraceOmited(self.span_trace)
+            SpanTraceOmited(self.span_trace.as_ref())
         )?;
         Ok(())
     }
 }
 
 #[cfg(feature = "capture-spantrace")]
-struct SpanTraceOmited<'a>(Option<&'a SpanTrace>);
+struct SpanTraceOmited<'a>(Option<&'a SpanTraceStatus>);
 
 #[cfg(feature = "capture-spantrace")]
 impl fmt::Display for SpanTraceOmited<'_> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        if let Some(span_trace) = self.0 {
-            if span_trace.status() == SpanTraceStatus::UNSUPPORTED {
-                writeln!(f, "Warning: SpanTrace capture is Unsupported.")?;
-                write!(
-                    f,
-                    "Ensure that you've setup an error layer and the versions match"
-                )?;
-            }
+        if let Some(&SpanTraceStatus::UNSUPPORTED) = self.0 {
+            writeln!(f, "Warning: SpanTrace capture is Unsupported.")?;
+            write!(
+                f,
+                "Ensure that you've setup an error layer and the versions match"
```

Comment:
I know it's unrelated to this PR, but I found this message confusing when I first encountered it. Perhaps some guidance as to how to "setup an error layer", and what exactly an "error layer" is would be helpful to users who are unfamiliar with the ecosystem.